demos/imagenet_classification_demo/include/grid_mat.hpp

```diff
@@ -0,0 +1,116 @@
+        cellSize.width = maxDisp.width * 1. / cNum;
+        cellSize.height = (maxDisp.height - rectangleHeight) * 1. / rNum;
+        
+        //size_t nGridCols = static_cast<size_t>(ceil(sqrt(static_cast<float>(sizes.size()))));
+        //size_t nGridRows = (sizes.size() - 1) / nGridCols + 1;
+
+        for (size_t i = 0; i < cNum * rNum; i++) {
+            cv::Point p;
+            p.x = cellSize.width * (i % cNum);
+            p.y = rectangleHeight + (cellSize.height * (i / cNum));
+            points.push_back(p);
+        }
+
+        outimg.create((cellSize.height * rNum) + rectangleHeight, cellSize.width * cNum, CV_8UC3);
+        outimg.setTo(0);
+        clear();
+    }
+
+    cv::Size getCellSize() {
+        return cellSize;
+    }
+
+    void listUpdate(std::queue<cv::Mat>& frames) {
+        while(!frames.empty()) {
+            updateList.push_back(frames.front());
+            frames.pop();
+        }
+    }
+
+    void update() {
+        while(!updateList.empty()) {    
+            cv::Mat cell = outimg(cv::Rect(points[currSourceID], cellSize));
+            cv::Mat frame = updateList.front();
+            updateList.pop_front();
+
+            if ((cellSize.width == frame.cols) && (cellSize.height == frame.rows)) {
+                frame.copyTo(cell);
+            } else if ((cellSize.width > frame.cols) && (cellSize.height > frame.rows)) {
+                frame.copyTo(cell(cv::Rect(0, 0, frame.cols, frame.rows)));
+            } else {
+                cv::resize(frame, cell, cellSize);
+            }
+            
+            if(currSourceID == points.size() - 1)
+                currSourceID = 0;
+            else
+                currSourceID++;
+        }
+    }
+
+    void textUpdate(double overSPF){
+        //set rectangle
+        size_t colunmNum = outimg.cols;
+        cv::Point p1 = cv::Point(0,0);
+        cv::Point p2 = cv::Point(colunmNum, rectangleHeight);
+        
+        rectangle(outimg,p1,p2,
+            cv::Scalar(0,0,0), cv::FILLED);
+        
+        //set text        
+        auto frameWidth = outimg.cols;
+        double fontScale = frameWidth * 1. / 640;
+        auto fontColor = cv::Scalar(0, 255, 0);
+        int thickness = 2;
+
+        cv::putText(outimg,
+                    cv::format("Overall FPS: %0.01f", 1./overSPF),
+                    cv::Point(10, static_cast<int>(30 * fontScale / 1.6)),
+                    cv::FONT_HERSHEY_PLAIN, fontScale, fontColor, thickness);
+    }
+
+    bool isFilled() const noexcept {
+        return unupdatedSourceIDs.empty();
+    }
+    void clear() {
+        size_t counter = 0;
+        std::generate_n(std::inserter(unupdatedSourceIDs, unupdatedSourceIDs.end()), points.size(), [&counter]{return counter++;});
+    }
+    std::set<size_t> getUnupdatedSourceIDs() const noexcept {
+        return unupdatedSourceIDs;
+    }
+    cv::Mat getMat() const noexcept {
+        return outimg;
```

Comment:
You can move `update()`'s body inside of `getMat()` and make `outimg` private. This will assure that the retrieved image is always up to date.
It should be `outImg` instead of `outimg`.